error due lack of continuing to support librsvg 2.40.x

Commit 1533e8def33bd3e1d7f4734206af4a66b51b9960 uses too new functions of librsvg

instead, let's add some conditional wheantime some people migrating softly so that it still builds with librsvg 2.40.x (which was the last versions fully in C) while we use newer code and no warnings when using newer versions

without that i got this error:

```
[   53s] src/icon.c: In function 'get_svg_icon':
[   53s] src/icon.c:104:2: error: unknown type name 'RsvgRectangle'; did you mean 'XRectangle'?
[   53s]   RsvgRectangle viewport = {
[   53s]   ^~~~~~~~~~~~~
[   53s]   XRectangle
[   53s] src/icon.c:105:3: error: field name not in record or union initializer
[   53s]    .x = 0,
[   53s]    ^
[   53s] src/icon.c:105:3: note: (near initialization for 'viewport')
[   53s] src/icon.c:106:3: error: field name not in record or union initializer
[   53s]    .y = 0,
[   53s]    ^
[   53s] src/icon.c:106:3: note: (near initialization for 'viewport')
[   53s] src/icon.c:106:8: warning: excess elements in scalar initializer
[   53s]    .y = 0,
[   53s]         ^
[   53s] src/icon.c:106:8: note: (near initialization for 'viewport')
[   53s] src/icon.c:107:3: error: field name not in record or union initializer
[   53s]    .width = size,
[   53s]    ^
[   53s] src/icon.c:107:3: note: (near initialization for 'viewport')
[   53s] src/icon.c:107:12: warning: excess elements in scalar initializer
[   53s]    .width = size,
[   53s]             ^~~~
[   53s] src/icon.c:107:12: note: (near initialization for 'viewport')
[   53s] src/icon.c:108:3: error: field name not in record or union initializer
[   53s]    .height = size,
[   53s]    ^
[   53s] src/icon.c:108:3: note: (near initialization for 'viewport')
[   53s] src/icon.c:108:13: warning: excess elements in scalar initializer
[   53s]    .height = size,
[   53s]              ^~~~
[   53s] src/icon.c:108:13: note: (near initialization for 'viewport')
[   53s] src/icon.c:122:2: warning: implicit declaration of function 'rsvg_handle_render_document'; did you mean 'rsvg_handle_render_cairo'? [-Wimplicit-function-declaration]
[   53s]   rsvg_handle_render_document(svg, cr, &viewport, &err);
[   53s]   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[   53s]   rsvg_handle_render_cairo
[   53s] make[1]: *** [Makefile:76: icon.o] Error 1
[   53s] make[1]: Leaving directory '/usr/src/packages/BUILD'
[   53s] dh_auto_build: make -j1 returned exit code 2
[   53s] make: *** [debian/rules:35: binary] Error 2
[   53s] dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
[   53s] ### VM INTERACTION START ###
[   53s] Failed to write utmp record: Read-only file system
[   53s] Powering off.
[   53s] [   46.304436][ T9720] reboot: Power down
[   53s] ### VM INTERACTION END ###
[   53s] 
[   53s] i02-ch2c failed "build jgmenu_4.4.1-1.dsc" at Tue Nov 14 15:20:15 UTC 2023.
[   53s] 
```

--- a/src/icon.c
+++ b/src/icon.c
@@ -104,12 +104,18 @@ static cairo_surface_t *get_svg_icon(const char *filename, int size)
 	cairo_t *cr;
 	RsvgHandle *svg;
 	GError *err = NULL;
+#if LIBRSVG_CHECK_VERSION(2, 52, 0)
 	RsvgRectangle viewport = {
 		.x = 0,
 		.y = 0,
 		.width = size,
 		.height = size,
 	};
+#else
+	RsvgDimensionData  dimensions;
+	double scale, ratio;
+#endif
+
 
 	svg = rsvg_handle_new_from_file(filename, &err);
 	if (err) {
@@ -119,6 +125,7 @@ static cairo_surface_t *get_svg_icon(const char *filename, int size)
 		return NULL;
 	}
 
+#if LIBRSVG_CHECK_VERSION(2, 52, 0)
 	surface = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, size, size);
 	cr = cairo_create(surface);
 
@@ -129,6 +136,29 @@ static cairo_surface_t *get_svg_icon(const char *filename, int size)
 		g_error_free(err);
 		return NULL;
 	}
+#else
+	rsvg_handle_get_dimensions(svg, &dimensions);
+	if (dimensions.width == dimensions.height) {
+		surface = cairo_image_surface_create(CAIRO_FORMAT_ARGB32,
+						     size, size);
+		cr = cairo_create(surface);
+		cairo_scale(cr, (double)size / dimensions.width,
+			    (double)size / dimensions.height);
+	} else if (dimensions.width > dimensions.height) {
+		ratio = (double)dimensions.width / dimensions.height;
+		scale = (double)size / dimensions.width;
+		surface = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, size, size / ratio);
+		cr = cairo_create(surface);
+		cairo_scale(cr, scale, scale);
+	} else {
+		ratio = (double)dimensions.width / dimensions.height;
+		scale = (double)size / dimensions.height;
+		surface = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, size * ratio, size);
+		cr = cairo_create(surface);
+		cairo_scale(cr, scale, scale);
+	}
+	rsvg_handle_render_cairo(svg, cr);
+#endif
 	cairo_destroy(cr);
 	g_object_unref(svg);
 

